src/main/java/com/diffblue/deeptestutils/powermock/WhiteboxImpl.java

```diff
@@ -0,0 +1,206 @@
+     *
+     * @param <T> the generic type
+     * @param type the class type
+     * @param methodsList list of methods to inspect
+     * @return list of methods declared in the provided class type with
+     *      removed duplicates, the original list otherwise
+     */
+    public static <T> List<Method> removeDuplicatesBasedOnReturnType(
+            final Class<T> type, final List<Method> methodsList) {
+        if (methodsList.size() < 2) {
+            return methodsList;
+        }
+
+        List<Method> methodsListTemp = new LinkedList<Method>();
+        for (Method method : methodsList) {
+            if (method.getDeclaringClass() == type) {
+                methodsListTemp.add(method);
+            }
+        }
+        switch (methodsListTemp.size()) {
+            case 0:
+                return methodsList;
+            case 1:
+                return methodsListTemp;
+            default:
+                Method candidate = methodsListTemp.get(0);
+                for (int i = 1; i < methodsListTemp.size(); i++) {
+                    if (candidate.getDeclaringClass().isAssignableFrom(
+                            methodsListTemp.get(i).getDeclaringClass())) {
+                        candidate = methodsListTemp.get(i);
+                    }
+                }
+                return Collections.singletonList(candidate);
+        }
+    }
+
+    /**
+     * Copied from org.powermock.reflect.internal.WhiteboxImpl
+     *
+     * Throw exception when multiple method matches found.
+     *
+     * @param helpInfo the help info
+     * @param methods  the methods
+     */
+    static void throwExceptionWhenMultipleMethodMatchesFound(
+            final String helpInfo,
+            final Method[] methods) {
+        if (methods == null || methods.length < 2) {
+            throw new IllegalArgumentException("Internal error: "
+                    + "throwExceptionWhenMultipleMethodMatchesFound "
+                    + "needs at least two methods.");
+        }
+        StringBuilder sb = new StringBuilder();
+        sb.append("Several matching methods found, please specify the ");
+        sb.append(helpInfo);
+        sb.append(" so that PowerMock can determine which method you're"
+                + " referring to.\n");
+        sb.append("Matching methods in class ").append(methods[0]
+                .getDeclaringClass().getName()).append(" were:\n");
+
+        for (Method method : methods) {
+            sb.append(method.getReturnType().getName()).append(" ");
+            sb.append(method.getName()).append("( ");
+            final Class<?>[] parameterTypes = method.getParameterTypes();
+            for (Class<?> paramType : parameterTypes) {
+                sb.append(paramType.getName()).append(".class ");
+            }
+            sb.append(")\n");
+        }
+        throw new TooManyMethodsFoundException(sb.toString());
+    }
+
+    /**
+     * Copied from org.powermock.reflect.internal.WhiteboxImpl and added few
```

Comment:
'few lines' is not very helpful. Explain the change in behaviour and the purpose of the change